File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    for i in range(1, brown+yellow+1):
        if (brown+yellow)%i == 0:
            a = (brown+yellow)//i
            if a >= i:
                if 2*a + 2*i == brown + 4:
                    return[a, i]
            
    
    return answer

File: 프로그래머스/lv2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        for (int i =1; i< brown+yellow ; i++){
            if( (brown+yellow) % i == 0){
                int y = (brown + yellow) / i;
                if( i<= y && (2*i + 2*y - 4) == brown){
                    answer[0] = y;
                    answer[1] = i;
                    return answer;
                }
            }
        }
        
        
        return answer;        
    }
}